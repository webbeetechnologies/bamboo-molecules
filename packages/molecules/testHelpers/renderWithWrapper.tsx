import type { ReactElement, JSXElementConstructor, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ProvideMolecules } from '../src';

export const renderWithWrapper = <P extends {} = {}>(
    component: ReactElement<P, string | JSXElementConstructor<P>>,
    options?: RenderOptions,
): ReturnType<typeof render> => {
    const Wrapper = ({ children }: { children: ReactNode }) => (
        <ProvideMolecules>{children}</ProvideMolecules>
    );

    return render(component, { wrapper: Wrapper, ...options });
};
