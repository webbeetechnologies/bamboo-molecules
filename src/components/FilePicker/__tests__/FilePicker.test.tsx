import { FilePicker } from '../../index';
import { renderWithWrapper } from '../../../../testHelpers';

jest.doMock('../../Icon/Icon', () => {
    return jest.fn(args => JSON.stringify(args));
});

const Icon = require('../../Icon/Icon');

it('renders FilePicker with default props', () => {
    const tree = renderWithWrapper(<FilePicker right={<Icon name="upload" />} />).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders outlined FilePicker', () => {
    const tree = renderWithWrapper(
        <FilePicker variant="outlined" right={<Icon name="upload" />} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders multiple images selectable FilePicker', () => {
    const tree = renderWithWrapper(
        <FilePicker multiple type={['image/*']} right={<Icon name="upload" />} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
