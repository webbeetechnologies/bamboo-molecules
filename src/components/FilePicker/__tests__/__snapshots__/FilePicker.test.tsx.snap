// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FilePicker with default props 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "animationScale": 1,
        "backgroundColor": "rgba(231, 224, 236, 1)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "flexDirection": "row",
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(73, 69, 79, 1)",
          "bottom": 0,
          "height": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "zIndex": 1,
        }
      }
      testID="text-input-underline"
    />
    <View
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
          "minHeight": 56,
          "paddingBottom": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        collapsable={false}
        onLayout={[Function]}
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        <Text
          collapsable={false}
          maxFontSizeMultiplier={15}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "rgba(73, 69, 79, 1)",
              "fontSize": 16,
              "fontWeight": "400",
              "left": 0,
              "lineHeight": 24,
              "opacity": 0,
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
            }
          }
          testID="text-input-label-active"
        >
          Choose file
        </Text>
        <Text
          collapsable={false}
          maxFontSizeMultiplier={15}
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(73, 69, 79, 1)",
              "fontSize": 16,
              "fontWeight": "400",
              "left": 0,
              "lineHeight": 24,
              "opacity": 1,
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
            }
          }
          testID="text-input-label-inactive"
        >
          Choose file
        </Text>
      </View>
      <TextInput
        editable={false}
        maxFontSizeMultiplier={15}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="rgba(73, 69, 79, 1)"
        size="md"
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
              "flexGrow": 1,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
              "paddingBottom": 8,
              "paddingTop": 24,
            },
            Object {
              "height": 56,
            },
            false,
            Object {
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            undefined,
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "color": "rgba(73, 69, 79, 1)",
          "iconSize": 24,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginRight": -4,
        }
      }
      testID="text-input-right-element"
    >
      {"name":"upload"}
    </View>
  </View>
</View>
`;

exports[`renders multiple images selectable FilePicker 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "animationScale": 1,
        "backgroundColor": "rgba(231, 224, 236, 1)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "flexDirection": "row",
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(73, 69, 79, 1)",
          "bottom": 0,
          "height": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "zIndex": 1,
        }
      }
      testID="text-input-underline"
    />
    <View
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
          "minHeight": 56,
          "paddingBottom": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        collapsable={false}
        onLayout={[Function]}
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        <Text
          collapsable={false}
          maxFontSizeMultiplier={15}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "rgba(73, 69, 79, 1)",
              "fontSize": 16,
              "fontWeight": "400",
              "left": 0,
              "lineHeight": 24,
              "opacity": 0,
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
            }
          }
          testID="text-input-label-active"
        >
          Choose file
        </Text>
        <Text
          collapsable={false}
          maxFontSizeMultiplier={15}
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(73, 69, 79, 1)",
              "fontSize": 16,
              "fontWeight": "400",
              "left": 0,
              "lineHeight": 24,
              "opacity": 1,
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
            }
          }
          testID="text-input-label-inactive"
        >
          Choose file
        </Text>
      </View>
      <TextInput
        editable={false}
        maxFontSizeMultiplier={15}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="rgba(73, 69, 79, 1)"
        size="md"
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
              "flexGrow": 1,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
              "paddingBottom": 8,
              "paddingTop": 24,
            },
            Object {
              "height": 56,
            },
            false,
            Object {
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            undefined,
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "color": "rgba(73, 69, 79, 1)",
          "iconSize": 24,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginRight": -4,
        }
      }
      testID="text-input-right-element"
    >
      {"name":"upload"}
    </View>
  </View>
</View>
`;

exports[`renders outlined FilePicker 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "animationScale": 1,
        "borderRadius": 4,
        "flexDirection": "row",
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 4,
          "borderWidth": 1,
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="text-input-outline"
    />
    <View
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
          "minHeight": 56,
          "paddingBottom": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        collapsable={false}
        onLayout={[Function]}
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        <Text
          collapsable={false}
          maxFontSizeMultiplier={15}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "rgba(73, 69, 79, 1)",
              "fontSize": 16,
              "fontWeight": "400",
              "left": 0,
              "lineHeight": 24,
              "marginLeft": -4,
              "opacity": 0,
              "paddingHorizontal": 4,
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
            }
          }
          testID="text-input-label-active"
        >
          Choose file
        </Text>
        <Text
          collapsable={false}
          maxFontSizeMultiplier={15}
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(73, 69, 79, 1)",
              "fontSize": 16,
              "fontWeight": "400",
              "left": 0,
              "lineHeight": 24,
              "marginLeft": -4,
              "opacity": 1,
              "paddingHorizontal": 4,
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
            }
          }
          testID="text-input-label-inactive"
        >
          Choose file
        </Text>
      </View>
      <TextInput
        editable={false}
        maxFontSizeMultiplier={15}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="rgba(73, 69, 79, 1)"
        size="md"
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
              "flexGrow": 1,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
              "paddingBottom": 4,
              "paddingTop": 4,
            },
            Object {
              "height": 56,
            },
            false,
            Object {
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            undefined,
          ]
        }
        testID="text-input-outlined"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "color": "rgba(73, 69, 79, 1)",
          "iconSize": 24,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginRight": -4,
        }
      }
      testID="text-input-right-element"
    >
      {"name":"upload"}
    </View>
  </View>
</View>
`;
